File: mylibrary/src/main/java/com/xiaolu/mylibrary/base/LazyFragment.java
package com.xiaolu.mylibrary.base;

import android.os.Bundle;

import com.xiaolu.mylibrary.mvpbase.BaseContract;
import com.xiaolu.mylibrary.mvpbase.BasePresenter;
@Deprecated
public abstract class LazyFragment<P extends BasePresenter<V>, V extends BaseContract.BaseView> extends BaseLazyFragment {
    protected P presenter;

    //实例化Presenter
    public abstract P initPresenter();

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        presenter = initPresenter();
        try {
            presenter.attachView((V) this);
            //checkActivityAttached();
            // initDate();
        } catch (Exception e) {
            new ClassCastException(this.toString() + "实现BasePresenter或者BaseView子类接口");
        }
    }

    @Override
    public void onDestroy() {
        if (presenter != null) {
            presenter.detachView();
        }
        super.onDestroy();
    }

    @Override
    public void onResume() {
        if (presenter != null) {
            presenter.attachView((V) this);
        } else {
            presenter = initPresenter();
        }
        super.onResume();
    }
}

File: mylibraryKotlin/src/main/java/com/xiaolu/mylibrarykotlin/base/LazyFragment.kt
package com.xiaolu.mylibrarykotlin.base

import android.os.Bundle
import com.xiaolu.mylibrarykotlin.mvpbase.BaseContract
import com.xiaolu.mylibrarykotlin.mvpbase.BasePresenter

@Deprecated("")
abstract class LazyFragment<P : BasePresenter<V>?, V : BaseContract.BaseView?> : BaseLazyFragment() {
    protected var presenter: P? = null

    //实例化Presenter
    abstract fun initPresenter(): P
    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        presenter = initPresenter()
        try {
            presenter!!.attachView(this as V)
            //checkActivityAttached();
            // initDate();
        } catch (e: Exception) {
            ClassCastException(this.toString() + "实现BasePresenter或者BaseView子类接口")
        }
    }

    override fun onDestroy() {
        if (presenter != null) {
            presenter?.detachView()
        }
        super.onDestroy()
    }

    override fun onResume() {
        if (presenter != null) {
            presenter?.attachView(this as V)
        } else {
            presenter = initPresenter()
        }
        super.onResume()
    }
}